api/types/entrySchema.go

```diff
@@ -0,0 +1,181 @@
+package apitypes
+
+import (
+	"encoding/json"
+	"fmt"
+	"sort"
+
+	"github.com/emirpasic/gods/maps/linkedhashmap"
+	"github.com/getlantern/deepcopy"
+	"github.com/puppetlabs/wash/plugin"
+)
+
+// EntrySchema describes an entry's schema, which
+// is what's returned by the /fs/schema endpoint.
+type EntrySchema struct {
+	plugin.EntrySchema
+	children map[string]*EntrySchema
+	// graph is an ordered map of <TypeID> => <EntrySchema>. We store it to make
+	// MarshalJSON's implementation easier.
+	//
+	// NOTE: The reason we don't synchronize children with graph is b/c entry
+	// schemas are immutable. Clients that want to mess with the state (e.g. like
+	// `wash find`) can do so at their own peril.
+	graph *linkedhashmap.Map
+}
+
+// MarshalJSON marshals the entry's schema to JSON. It takes
+// a value receiver so that the entry schema's still marshalled
+// when it's referenced as an interface{} object. See
+// https://stackoverflow.com/a/21394657 for more details.
+func (s EntrySchema) MarshalJSON() ([]byte, error) {
+	if s.graph != nil {
+		return s.graph.ToJSON()
+	}
+	return json.Marshal(s.EntrySchema)
+}
+
+// UnmarshalJSON unmarshals the entry's schema from JSON.
+func (s *EntrySchema) UnmarshalJSON(data []byte) error {
+	rawGraph := linkedhashmap.New()
+	if err := rawGraph.FromJSON(data); err != nil {
+		return err
+	}
+	if rawGraph.Size() <= 0 {
+		msg := fmt.Sprintf("s.UnmarshalJSON: expected a non-empty JSON object but got %v instead", string(data))
+		panic(msg)
+	}
+	s.graph = linkedhashmap.New()
+	s.children = make(map[string]*EntrySchema)
```

Comment:
This isn't filled in. Should it be?

Reply:
That's filled in at line 80. Reason I created the map here was to avoid a nil pointer exception since I'm creating it for the other cases at line 66. But I think I can clean that up a bit. Will update.